apollo-runtime/src/main/java/com/apollographql/android/cache/CacheInterceptor.java

```diff
@@ -121,6 +130,22 @@ private Response withServedDateHeader(Response response) throws IOException {
         .build();
   }
 
+  private Response prefetch(Response networkResponse, String cacheKey) throws IOException {
+    cache.write(networkResponse, cacheKey);
+    Response cacheResponse = cache.read(cacheKey);
+    if (cacheResponse == null) {
+      throw new IOException("failed to read prefetch cache response");
+    }
+    return cacheResponse
+        .newBuilder()
+        .networkResponse(strip(networkResponse))
+        .build();
+  }
+
+  private static boolean isPrefetchResponse(Request request) {
+    return Boolean.TRUE.toString().equals(request.header(HttpCache.CACHE_PREFETCH_HEADER));
+  }
+
```

Comment:
General note that this class is getting a bit long. Would be nice to move some stuff to a utility or some delegate